api/handler/tacoma.go

```diff
@@ -0,0 +1,95 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"spd-lookup/api/data"
+	"strings"
+)
+
+// TacomaOfficerMetadata is the handler function for retrieving Tacoma metadata
+func (h *Handler) TacomaOfficerMetadata(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(h.db.TacomaOfficerMetadata())
+}
+
+// TacomaStrictMatch is the handler function for retrieving Tacoma officers with a strict match
+func (h *Handler) TacomaStrictMatch(w http.ResponseWriter, r *http.Request) {
+	badge, firstName, lastName := r.URL.Query().Get("badge"), strings.TrimSpace(r.URL.Query().Get("first_name")), strings.TrimSpace(r.URL.Query().Get("last_name"))
+
+	if badge != "" && firstName == "" && lastName == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("At this time we do not have the badge numbers available for Tacoma PD. Please attempt searches by first or last name only.")))
+		return
+	}
+
+	if firstName == "" && lastName == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("at least one of the following parameters must be provided: first_name, last_name")))
+		return
+	}
+
+	if firstName == "" {
```

Comment:
I don't know enough about Go, but it seems like there are a couple of blocks of code (or whole functions) that are shared between Seattle and Tacoma due to the similarity of datasets. Would it be possible to have a set of shared functions (like for sorting, string replacement logic, etc) that could be used across all datasets?

Reply:
imo since we're doing this as a way to break up and handle dissimilar data sets it makes sense to leave these separate like this. It's pretty verbose this way, but, unfortunately, I think this is the safest way to handle it so we don't have to extract each data sets separate functions from each other down the line because it doesn't line up. It makes sense that first and last names will always be somewhere in there, but in the event we only have a set of badge numbers or something similar, I think this is the safest route.

Reply:
yea, exactly what @mpuckett159 said. a common complaint about go is its verbosity, but the introduction of other departments hopefully would be mostly copy and paste.

the bit about first/last name is interesting too. if we can make that assumption, we could create a base officer struct of just first and last name, and extend (or embed) that into department specific structs

Reply:
Gotcha, I guess we'll have to reevaluate this once we start adding more datasets!